numbat/src/bytecode_interpreter.rs

```diff
@@ -165,6 +167,31 @@ impl BytecodeInterpreter {
                     self.vm.add_op2(Op::Call, idx, args.len() as u16); // TODO: check overflow
                 }
             }
+            Expression::MakeStruct(_span, exprs, struct_info) => {
+                // structs must be consistently ordered in the VM, so we reorder
+                // the field values so that they are evaluated in the order the
+                // struct fields are defined.
+
+                let sorted_exprs = exprs
+                    .iter()
+                    .sorted_by_key(|(n, _)| struct_info.fields.get_index_of(n).unwrap());
+
+                for (_, expr) in sorted_exprs.rev() {
+                    self.compile_expression_with_simplify(expr)?;
+                }
+
+                let struct_info_idx = self.vm.get_structinfo_idx(&struct_info.name).unwrap() as u16;
+
+                self.vm
+                    .add_op2(Op::BuildStruct, struct_info_idx, exprs.len() as u16);
```

Comment:
Similar here: maybe rename the operation to `BuildStructInstance` or just `BuildInstance`